src/Catalyst.Benchmark/Catalyst.Abstractions/CryptoBenchmark.cs

```diff
@@ -0,0 +1,111 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Jobs;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Protocol.Cryptography;
+using Catalyst.Protocol.Network;
+using Catalyst.Protocol.Transaction;
+using Catalyst.Protocol.Wire;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+
+namespace Catalyst.Benchmark.Catalyst.Core.Modules.Hashing
+{
+    [MemoryDiagnoser]
+    [SimpleJob(RuntimeMoniker.CoreRt30)]
+    public class CryptoBenchmark
+    {
+        static CryptoBenchmark()
+        {
+            Context = new SigningContext
+            {
+                NetworkType = NetworkType.Mainnet,
+                SignatureType = SignatureType.TransactionConfidential
+            };
+
+            var key = ByteString.CopyFrom(new byte[32]);
+            var amount = ByteString.CopyFrom(new byte[32]);
+
+            Transaction = new TransactionBroadcast
+            {
+                PublicEntries =
+                {
+                    new PublicEntry
+                    {
+                        Amount = amount,
+                        Base = new BaseEntry
+                        {
+                            Nonce = 1,
+                            SenderPublicKey = key,
+                            ReceiverPublicKey = key,
+                            TransactionFees = amount
+                        }
+                    }
+                },
+                Timestamp = Timestamp.FromDateTime(DateTime.UtcNow),
+            };
+
+            Crypto = new NoopCryptoContext();
+        }
+
+        static readonly SigningContext Context;
+        static readonly TransactionBroadcast Transaction;
+        static readonly ICryptoContext Crypto;
+        static readonly IPrivateKey PrivateKey = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this initialization to 'PrivateKey', the compiler will do that for you.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4569935)